### PR TITLE
New blog: Deploy and 'take Flyte' with an end-to-end ML orchestration solution on AKS

### PR DESCRIPTION
**Pull request by sdesai345**
This blog was drafted in collaboration with folks from Union.ai, making improvements to enable cluster auto-upgrade and monitoring to their default reference implementation on Azure.

### CODE REVIEW

File: blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+date: 2024-11-20
+author: Sachi Desai
+categories: ai
+---
+
+## Background
+
+Data is often at the heart of application design and development - it fuels user-centric design, provides insights for feature enhancements, and represents the value of an application as a whole. In that case, shouldn’t we use data science tools and workflows that are flexible and scalable on a platform like Kubernetes, for a range of application types?
+
+In collaboration with [David Espejo](https://www.linkedin.com/in/davidmirror/) and [Shalabh Chaudhri](https://www.linkedin.com/in/shalabhchaudhri/) from [Union.ai](https://www.union.ai/), we’ll dive into an example using *Flyte*, a platform built on Kubernetes itself. Flyte can help you manage and scale out data processing and machine learning pipelines through a simple user interface.
```
Comment (pauldotyu) [inline blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md:13]: Should we add a link to Flyte here?

```suggestion
In collaboration with [David Espejo](https://www.linkedin.com/in/davidmirror/) and [Shalabh Chaudhri](https://www.linkedin.com/in/shalabhchaudhri/) from [Union.ai](https://www.union.ai/), we’ll dive into an example using [Flyte](https://flyte.org/), a platform built on Kubernetes itself. Flyte can help you manage and scale out data processing and machine learning pipelines through a simple user interface.
```

File: blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+> To create a GPU-enabled node pool using your `aks.tf` [configuration file](https://github.com/unionai-oss/deploy-flyte/blob/main/environments/azure/flyte-core/aks.tf), update the `gpu_node_pool_count` and `gpu_machine_type` to your desired node count and instance type, respectively, in the `locals` array. If you specify an `accelerator` type, make sure to select a [supported option](https://github.com/flyteorg/flytekit/blob/daeff3f5f0f36a1a9a1f86c5e024d1b76cdfd5cb/flytekit/extras/accelerators.py#L132-L160) for flytekit.
+
+### Identity management
+
+[Workload Identity](https://learn.microsoft.com/azure/aks/workload-identity-overview) Federation with Entra ID enables fine-grained security controls to be applied. The `flytepropeller`, `flyteadmin`, and `datacatalog` backend components use one user-assigned MI, while the Flyte task execution pods use a separate user-assigned MI.
```
Comment (pauldotyu) [inline blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md:87]:
```suggestion
[Workload Identity](https://learn.microsoft.com/azure/aks/workload-identity-overview) Federation with Entra ID enables fine-grained security controls to be applied. The `flytepropeller`, `flyteadmin`, and `datacatalog` backend components use one user-assigned MI, while the Flyte task execution pods use a separate user-assigned managed identity (MI).
```

File: blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+## Let's get started
+
+Before you begin, take a look at the [prerequisites](https://github.com/unionai-oss/deploy-flyte/blob/main/environments/azure/flyte-core/README.md#prerequisites) on Azure, including:
+
+* Azure subscription with at least [Contributor role over all your resources](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles/privileged#contributor) using Azure RBAC.
+* Azure CLI version 2.0 or later installed and configured.
+* Terraform version 1.3.7 or later installed.
+* Helm version 3.15.4 or later installed.
+* Kubernetes command-line client, [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/), installed and configured.
+
```
Comment (sdesai345) [inline blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md:45]:
```suggestion

Now, you can deploy the dependencies and install Flyte following [these steps](https://github.com/unionai-oss/deploy-flyte/blob/main/environments/azure/flyte-core/README.md) and come back to this blog for a deep dive of your solution on AKS!

```

**Review by chzbrgr71 — APPROVED** [review]
Looks good. I previously walked through this on the first draft and this is ready to publish.

File: blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
+---
+title: "Deploy and 'take Flyte' with an end-to-end ML orchestration solution on AKS"
```
Comment (pauldotyu) [inline blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md:2]:
```suggestion
title: "Deploy and take Flyte with an end-to-end ML orchestration solution on AKS"
```

File: blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+![image](/AKS/assets/images/deploy-data-science-solution-with-flyte/successful-flyte-deployment-on-aks.png)
+
+
+## What next?
+
+* Check out more ML and feature engineering [tutorials with Flyte](https://docs.flyte.org/en/latest/flytesnacks/tutorials/index.html)
+* Learn about [best practices for your MLOps pipelines](https://learn.microsoft.com/azure/aks/best-practices-ml-ops) on Azure Kubernetes Service (AKS)
+* Explore the Flyte GitHub repository: https://github.com/flyteorg/flyte 
+* Join the #flyte-on-azure Slack channel: https://slack.flyte.org
```
Comment (pauldotyu) [inline blog/_posts/2024-11-20-e2e-flyte-ML-solution-on-aks.md:133]:
```suggestion
* Explore the Flyte GitHub repository: [https://github.com/flyteorg/flyte](https://github.com/flyteorg/flyte)
* Join the [#flyte-on-azure](https://slack.flyte.org) Slack channel: [https://slack.flyte.org](https://slack.flyte.org)
```

**Review by pauldotyu — APPROVED** [review]

**Pull request merged**